Makefile

```diff
@@ -1,22 +1,72 @@
+# ========================
+# Shared Variables
+# ========================
+BUILD_TYPE ?= debug
+BUILD_DIR := build/$(BUILD_TYPE)
+BUILD_SRC_RUST := build/src/rust_extensions
+CURRENT_BUILD := build/current
+DELTA_DIR := rust_extensions/delta
+DELTA_HEADER := $(DELTA_DIR)/target/cxxbridge/delta/src/lib.rs.h
+DELTA_LIB := $(DELTA_DIR)/target/$(BUILD_TYPE)/libdelta.a
+DUCKDB_DIR := third_party/duckdb
+DUCKDB_BUILD := $(DUCKDB_DIR)/build/$(BUILD_TYPE)/src/libduckdb.so
+SRC_DIR := src
+
+# ========================
+# Flags
+# ========================
 export override DEBUG := $(filter debug,$(BUILD_TYPE))
 CARGO_FLAGS := $(if $(DEBUG),,--release)
 MAKEFLAGS := --no-print-directory
 
-.PHONY: .BUILD all release debug clean format install installcheck uninstall \
-        duckdb duckdb-fast clean-duckdb \
-        delta clean-delta format-delta
-
-.BUILD:
+# ========================
+# Phony Targets
+# ========================
+.PHONY: .BUILD all clean clean-all clean-delta clean-duckdb debug delta \
+        duckdb duckdb-fast format format-delta help install installcheck \
+        release uninstall
+
+# ========================
+# Default Target: Help
+# ========================
+help:
+	@echo "Usage: make [TARGET] [BUILD_TYPE=debug|release]"
```

Comment:
Users should use `make debug` and `make release` for regular builds, rather than manually specifying `BUILD_TYPE`
The `BUILD_TYPE` variable is intended for advanced and rare cases where users need to build specific components (e.g., `delta`, `duckdb`) directly
As such, `BUILD_TYPE` should not appear in the main help line

Reply:
I set debug as the default type [here](https://github.com/Mooncake-Labs/pg_mooncake/pull/59/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R4) to prevent the user from having to specify `DEBUG_TYPE` in cases outside of `make debug/release`. We can remove this from the usage line if you think it will cause confusion, but it might be wise to leave [this](https://github.com/Mooncake-Labs/pg_mooncake/pull/59/files/48c72b30ae010585bbfc042c68a4ce0faa273c14#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R35) for transparency.

Reply:
See my new comment https://github.com/Mooncake-Labs/pg_mooncake/pull/59#discussion_r1899045480
I think it’s better to say that those targets are internal and not meant to be called directly. This way, end users don’t need to be aware of them or `DEBUG_TYPE`